pallene/checker.lua

```diff
@@ -636,29 +638,47 @@ function FunChecker:check_var(var)
         elseif cname._tag == "checker.Name.Function" then
             var._type = self.p.module.functions[cname.id].typ
         elseif cname._tag == "checker.Name.Builtin" then
-            var._type = builtins[cname.name].typ
+            var._type = builtins.functions[cname.name].typ
         elseif cname._tag == "checker.Name.Module" then
             -- The type for a module is a table.
-            var._type = builtin_modules[cname.name]
+           var._type = builtins.modules[cname.name]
```

Comment:
This case should be a type error, just like `checker.Name.Type`. This way, we will disallow things such as

    local x = io

By the way, don't forget to add a test case for that.

Reply:
Alright. Will start working on it tomorrow.